### PR TITLE
Entirely removes BIND_VMETHOD in favor of GDVIRTUAL

### PR DESCRIPTION
**Pull request by reduz**
* `_gui_input`, `_input`, `_unhandled_input` and `_unhandled_key_input` are now regular C++ virutal functions.
* Everything else converted to GDVIRTUAL
* BIND_VMETHOD is gone, always use the new syntax from now on.

Creating `_gui_input` method and using the binder to register events will no longer work, simply override the virtual function now.

### CODE REVIEW

File: editor/plugins/animation_player_editor_plugin.cpp [hunk 1]
```diff
@@ -1220,7 +1220,7 @@ void AnimationPlayerEditor::_onion_skinning_menu(int p_option) {
 	}
 }
 
-void AnimationPlayerEditor::_unhandled_key_input(const Ref<InputEvent> &p_ev) {
+void AnimationPlayerEditor::unhandled_key_input(const Ref<InputEvent> &p_ev) {
```
Comment (aaronfranke) [inline editor/plugins/animation_player_editor_plugin.cpp:1223]: Should this be `Ref<InputEventKey`>?

Comment (reduz) [reply editor/plugins/animation_player_editor_plugin.cpp:1223]: No.. at some point this was changed to also receive InputEventShurtcut too, so I updated the documentation accordingly.

File: core/register_core_types.cpp [hunk 2]
```diff
@@ -145,10 +146,12 @@ void register_core_types() {
 	GDREGISTER_CLASS(Resource);
 	GDREGISTER_CLASS(Image);
 
+	GDREGISTER_CLASS(Shortcut);
 	GDREGISTER_VIRTUAL_CLASS(InputEvent);
 	GDREGISTER_VIRTUAL_CLASS(InputEventWithModifiers);
 	GDREGISTER_VIRTUAL_CLASS(InputEventFromWindow);
 	GDREGISTER_CLASS(InputEventKey);
+	GDREGISTER_CLASS(InputEventShortcut);
```
Comment (KoBeWi) [inline core/register_core_types.cpp:154]: Exposing InputEventShortcut is a bit useless without exposing the `shortcut` property.

File: doc/classes/Node.xml [hunk 3]
```diff
@@ -43,14 +43,6 @@
...
 			</description>
 		</method>
-		<method name="_get_configuration_warnings" qualifiers="virtual">
-			<return type="String[]" />
-			<description>
-				The elements in the array returned from this method are displayed as warnings in the Scene Dock if the script that overrides it is a [code]tool[/code] script.
-				Returning an empty array produces no warnings.
-				Call [method update_configuration_warnings] when the warnings need to be updated for this node.
-			</description>
-		</method>
```
Comment (KoBeWi) [inline doc/classes/Node.xml:53]: What happened to this method description? It's being completely removed.

File: scene/main/node.cpp [hunk 4]
```diff
@@ -2535,6 +2540,35 @@ void Node::request_ready() {
...
+	}
+	unhandled_key_input(p_event);
+}
+
+void Node::input(const Ref<InputEvent> &p_event) {
+}
+void Node::unhandled_input(const Ref<InputEvent> &p_event) {
+}
+void Node::unhandled_key_input(const Ref<InputEvent> &p_key_event) {
+}
```
Comment (KoBeWi) [inline scene/main/node.cpp:2570]:
```suggestion
void Node::input(const Ref<InputEvent> &p_event) {
}

void Node::unhandled_input(const Ref<InputEvent> &p_event) {
}

void Node::unhandled_key_input(const Ref<InputEvent> &p_key_event) {
}
```
™
(same above)

File: modules/visual_script/visual_script.cpp [hunk 5]
```diff
@@ -1845,26 +1845,6 @@ void VisualScriptInstance::create(const Ref<VisualScript> &p_script, Object *p_o
...
-			node->set_process_input(true);
-		}
-		if (p_script->functions.has("_unhandled_input")) {
-			node->set_process_unhandled_input(true);
-		}
-		if (p_script->functions.has("_unhandled_key_input")) {
-			node->set_process_unhandled_key_input(true);
-		}
-	}
-
```
Comment (KoBeWi) [inline modules/visual_script/visual_script.cpp:1867]: This involves VisualScript (see `p_script` parameter), so the logic is probably different here and it doesn't need to be removed?

Comment (reduz) [reply modules/visual_script/visual_script.cpp:1867]: Logic should be the same. Visual Script was created in the 2.x branch where you had to set processing for these manually, so Visual Script did automatically. Nowadays the same exact logic exists in NOTIFICATION_READY of Node, making this redundant.

File: scene/main/node.cpp (hunk 4 above)
Comment (reduz) [reply scene/main/node.cpp:2570]: good catch, whish we could handle this automatically

File: core/register_core_types.cpp (hunk 2 above)
Comment (reduz) [reply core/register_core_types.cpp:154]: done

File: doc/classes/Node.xml (hunk 3 above)
Comment (reduz) [reply doc/classes/Node.xml:53]: The function existed twice due to a mistake from my part. The actual one is above this.

**Pull request merged**